### PR TITLE
etcdserver: add failpoints for backend

### PR DESCRIPTION
**Pull request by ahrtr**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

cc @ptabor @serathius @spzala

### CODE REVIEW

**Comment by ahrtr** [comment]
part of https://github.com/etcd-io/etcd/issues/14735

File: tests/linearizability/failpoints.go [hunk 1]
```diff
@@ -31,19 +31,25 @@ import (
...
+	BeforeCommitPanic               Failpoint = goFailpoint{"backend/beforeCommit", "panic", nil}
+	AfterCommitPanic                Failpoint = goFailpoint{"backend/afterCommit", "panic", nil}
+	RaftBeforeSavePanic             Failpoint = goFailpoint{"etcdserver/raftBeforeSave", "panic", nil}
+	RaftAfterSavePanic              Failpoint = goFailpoint{"etcdserver/raftAfterSave", "panic", nil}
+	BackendBeforePreCommitHookPanic Failpoint = goFailpoint{"backend/commitBeforePreCommitHook", "panic", nil}
+	BackendAfterPreCommitHookPanic  Failpoint = goFailpoint{"backend/commitAfterPreCommitHook", "panic", nil}
+	BackendBeforeStartDBTxnPanic    Failpoint = goFailpoint{"backend/beforeStartDBTxn", "panic", nil}
+	BackendAfterStartDBTxnPanic     Failpoint = goFailpoint{"backend/afterStartDBTxn", "panic", nil}
+	BackendBeforeWritebackBufPanic  Failpoint = goFailpoint{"backend/beforeStartDBTxn", "panic", nil}
+	BackendAfterWritebackBufPanic   Failpoint = goFailpoint{"backend/afterStartDBTxn", "panic", nil}
```
Comment (serathius) [inline tests/linearizability/failpoints.go:46]:
```suggestion
	BackendAfterWritebackBufPanic   Failpoint = goFailpoint{"backend/afterWritebackBufPanic", "panic", nil}
```

File: tests/linearizability/failpoints.go [hunk 2]
```diff
@@ -31,19 +31,25 @@ import (
...
+	DefragBeforeRenamePanic         Failpoint = goFailpoint{"backend/defragBeforeRename", "panic", triggerDefrag}
+	BeforeCommitPanic               Failpoint = goFailpoint{"backend/beforeCommit", "panic", nil}
+	AfterCommitPanic                Failpoint = goFailpoint{"backend/afterCommit", "panic", nil}
+	RaftBeforeSavePanic             Failpoint = goFailpoint{"etcdserver/raftBeforeSave", "panic", nil}
+	RaftAfterSavePanic              Failpoint = goFailpoint{"etcdserver/raftAfterSave", "panic", nil}
+	BackendBeforePreCommitHookPanic Failpoint = goFailpoint{"backend/commitBeforePreCommitHook", "panic", nil}
+	BackendAfterPreCommitHookPanic  Failpoint = goFailpoint{"backend/commitAfterPreCommitHook", "panic", nil}
+	BackendBeforeStartDBTxnPanic    Failpoint = goFailpoint{"backend/beforeStartDBTxn", "panic", nil}
+	BackendAfterStartDBTxnPanic     Failpoint = goFailpoint{"backend/afterStartDBTxn", "panic", nil}
+	BackendBeforeWritebackBufPanic  Failpoint = goFailpoint{"backend/beforeStartDBTxn", "panic", nil}
```
Comment (serathius) [inline tests/linearizability/failpoints.go:45]:
```suggestion
	BackendBeforeWritebackBufPanic  Failpoint = goFailpoint{"backend/beforeWritebackBufPanic", "panic", nil}
```

File: tests/linearizability/failpoints.go (hunk 1 above)
Comment (ahrtr) [reply tests/linearizability/failpoints.go]: thx. It's a copy-paste typo

**Comment by ahrtr** [comment]
Done. PTAL.

**Review by serathius — APPROVED** [review]

**Pull request merged**